modules-packages-crates/includes/1-scoping-terminology.md

```diff
@@ -0,0 +1,101 @@
+# Scoping Terminology
+
+Before we start, it is important to explain the concepts behind code organization in Rust programs:
+
+- A **package** is one or more **crates** that provide a set of functionality. A **package** contains a `Cargo.toml` file that describes how to build those **crates**.
+- A **crate** is a compilation unit, which is the smallest amount of code that the Rust compiler can operate on. Once compiled, it will produce either an executable or a library.
+- A **crate** contains an implicit, un-named top-level **module**.
+- A **module** is a *(possibly nested)* unit of code organization inside a **crate**.
+- Recursive definitions can span **modules**, but not **crates**.
+
+## Package
+
+Whenever we run the command `$ cargo new <project-name>`, Cargo creates a package for us:
+
+    $ cargo new my-project
+         Created binary (application) `my-project` package
+
+Here, we have a package that only contains `src/main.rs`, meaning it only contains a binary crate named `my-project`:
+
+    my-project
+    ├── src
+    │  └── main.rs
+    └── Cargo.toml
+
+A package can have multiple binary crates by placing files in the `src/bin` directory: each file will be a separate binary crate.
+
+If a package contains `src/main.rs` and `src/lib.rs`, it has two crates: a library and a binary, both with the same name as the package.
+
+## Crates
+
+Rust's compilation model centers on artifacts called crates, that can be compiled into a binary or into a library.
```

Comment:
Are there any examples of crates you can share? Just thinking of a little more detail to add to this section.